### ISSUE TITLE
Slow UI update when changing movies

### ISSUE THREAD

**Issue by fungos** [Komet/MediaElch#1640]
**Describe the bug**
Changing movie selection for movies that did not load from database yet, is very slow.

**To Reproduce**
1. Go to 'Movies' screen
2. Click on any movie that is not on DB (has a Star before its name)
3. Click on any other movie that is also not on DB or loaded before
4. See how long it takes to change from one movie to another

**Expected behavior**
Fast responsive UI

**MediaElch Version:**
 - [X] 2.10.5-dev (nightly)
 - [ ] 2.10.4 stable

**Operating System:**
 - [ ] Windows
 - [ ] macOS
 - [X] Linux (distribution: Manjaro)

**Additional context**
![image](https://github.com/Komet/MediaElch/assets/151544/597c9ec6-0af3-4272-9438-6f911e90c537)
![image](https://github.com/Komet/MediaElch/assets/151544/afd1d689-7033-466a-bb57-f8a330c0ee8b)

Doing this change improves a little, still not fast enough. The issue is the amount of widgets that will receive and process the event `dataChanged`

**Comment by bugwelle** [comment]
This has been in MediaElch right from the beginning and has annoyed me quite a bit :smile:

The amount of "sigChanged" signals is _a lot_ when a movie is selected. Adding a log entry:

```
2023-09-07 11:37:13.070 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: false
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.084 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.085 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.086 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.087 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: true
2023-09-07 11:37:13.088 DEBUG [../MediaElch/src/data/movie/Movie.cpp:791] | sigChanged: false
2023-09-07 11:37:13.331 DEBUG [../MediaElch/src/ui/movies/MovieWidget.cpp:335] | [MovieWidget] Opening: "Ghostbusters"
```

That only happens the first time a movie is opened, though (all images are loaded into cache, etc.)

I'll see what we can do. What needs to be checked:

- Can we change to a "commit()" method? Or at least reduce the number of signals?
- Are images loaded asynchronously or blocking?

Thanks for reporting.

Regards,
Andre

**Comment by bugwelle** [comment]
Should be fixed by #1643. Took a while, but I finally got to add caching to #1643.

**Comment by bugwelle** [comment]
Wow, just checked an older version on my Windows 10 instance. It was extremely slow! No idea why, though. With the Qt5 version (for Windows 7/8), it was a lot faster.
